src/program.js

```diff
@@ -251,6 +251,12 @@ Example: $0 --help run.
         demand: false,
         type: 'boolean',
       },
+      'pref': {
+        describe: 'Launch firefox with custom preferences. Lightweight ' +
+                  'alternative to creating custom profile.',
+        demand: false,
+        type: 'object',
```

Comment:
After reading doc for some time I understand that it would look like `"pref.dotted.name=value"` and years will automatically turn this into `pref = {dotted: {name:value}}`, hence repeated option like `"pref.name=value pref.anotherName=value"` will just be a nested object handled my yargs. That's really cool! 
However I'm still not sure about when the type detection is happening. Numbers should be detected automatically as the link you've shared suggests, as I understand coercing has those capabilities and we don't need to know in advance which properties are used http://yargs.js.org/docs/#methods-coercekey-fn. I imagine it looks g something like this
 ```
yargs.coerce('pref', function (obj) {
  for (var in obj) {
    if (!isNaN(Number(obj.var))) {
      obj.var == Number(obj.var);
    } else if (obj.var === 'true' || obj.var === 'false) {
       obj.var == Boolean(obj.var);
    }
  }
})
```
Let me know if that is what you had in mind